src/convertToObject.js

```diff
@@ -3,16 +3,39 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are
+ * the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const parts = sourceString
+    .replace(/:/g, ';')
+    .split(';')
+    .map(value => value
+      .split('')
+      .filter(symbol => symbol !== '\n')
+      .join('').trim())
+    .filter((element) => element !== '');
+
+  const keys = parts
+    .filter(value => parts.indexOf(value) % 2 === 0);
+  const values = parts
+    .filter(value => parts.indexOf(value) % 2 !== 0);
```

Comment:
Looks strange. Too many loops for the current task.
Also, in the filter method you have access to the index and the current array, so `parts.indexOf(value)` makes no sense.
`filter((value, index, array) => index % 2 === 0);`
Docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter